src/components/modal/z-modal-login/index.tsx

```diff
@@ -0,0 +1,233 @@
+import {
+  Component,
+  h,
+  Element,
+  Event,
+  EventEmitter,
+  Prop,
+} from "@stencil/core";
+import { HostElement, State } from "@stencil/core/internal";
+import {
+  ButtonVariantEnum,
+  InputStatusBean,
+  InputStatusEnum,
+} from "../../../beans";
+import { getSiblings } from "../../../utils/utils";
+
+/**
+ * @slot username - username input
+ * @slot password - password input
+ * @slot login - login button
+ * @slot signup - signup button
+ * @slot provider - expternal providers login buttons
+ */
+@Component({
+  tag: "z-modal-login",
+  styleUrl: "styles.css",
+  shadow: false,
+  scoped: true,
+})
+export class zModalLogin {
+  @Element() hostElement: HostElement;
+
+  /** Login modal title */
+  @Prop() heading?: string = "Entra in MyZanichelli";
+  /** Username/password input status */
+  @Prop({ mutable: true }) status?: InputStatusBean;
+  /** Username helper message */
+  @Prop({ mutable: true }) message?: string;
+
+  @State() externalProviderCheck: boolean = false;
+
+  componentDidLoad() {
+    this.externalProviderCheck = !!getSiblings(
```

Comment:
this seems a little too "tricky" to me 🤔 
In the `z-menu` [PR](https://github.com/ZanichelliEditore/design-system/pull/95/files#diff-1c46c07cd3ceb06540226ab084d9e81c723276b1b4bc339435142b0d2da70659R71) I used something like this `this.hostElement.querySelectorAll('[slot="provider"]')` and it worked fine. It returns all the elements slotted as "provider", so you don't have to use `getSiblings` or check weird things (`sibl.nodeName !== "Z-BODY"`) 🙂

Reply:
@ufollettu (he's the author of that piece of code) how about this? It works!

Reply:
LGTM